src/e2e/puppeteer/__tests__/browserBack.ts

```diff
@@ -0,0 +1,53 @@
+import clickThought from '../helpers/clickThought'
+import paste from '../helpers/paste'
+import press from '../helpers/press'
+import waitForEditable from '../helpers/waitForEditable'
+import { page } from '../setup'
+
+vi.setConfig({ testTimeout: 20000 })
+
+it('browser back button should work correctly after cursor movements', async () => {
+  const importText = `
+  - a
+  - b`
+
+  await paste(importText)
+
+  await waitForEditable('b')
+  await clickThought('a')
+  await press('ArrowDown')
+  await page.goBack()
+  await waitForEditable('a')
+  await press('ArrowDown')
+  await page.goBack()
+  const editable = await waitForEditable('a')
```

Comment:
`waitForEditable` just finds a thought on the page, not the cursor thought.

You can use https://github.com/cybersemics/em/blob/16fd8ed8e1bb6d54e6c0a8253a9e9e15ff7b1e40/src/e2e/puppeteer/helpers/getEditingText.ts to find the cursor thought.